src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Runtimes/MissingSdkRuntimeDetector.cs

```diff
@@ -48,32 +53,36 @@ protected override Task DisposeCoreAsync(bool initialized)
-            // safe to use within IProjectDynamicLoadComponent.LoadAsync.
             _projectGuid = await _project.UnconfiguredProject.GetProjectGuidAsync();
             _missingSetupComponentRegistrationService.RegisterProjectConfiguration(_projectGuid, _project);
-            _ = RegisterSdkRuntimeNeededInProjectAsync(_project);
+            _ = RegisterSdkNetCoreRuntimeNeededInProjectAsync(_project);
         }
 
-        private async Task RegisterSdkRuntimeNeededInProjectAsync(ConfiguredProject project)
+        private async Task RegisterSdkNetCoreRuntimeNeededInProjectAsync(ConfiguredProject project)
         {
             if (_enabled)
             {
-                var projectProperties = project.Services.ExportProvider.GetExportedValue<ProjectProperties>();
+                (string targetFramework, string targetFrameworkVersion) = await GetTargetFrameworkAsync(project);
+                targetFrameworkVersion = IsNotNetCore(targetFramework, targetFrameworkVersion) ? string.Empty : targetFrameworkVersion;
 
-                ConfigurationGeneral configuration = await projectProperties.GetConfigurationGeneralPropertiesAsync();
+                _missingSetupComponentRegistrationService.RegisterPossibleMissingSdkRuntimeVersion(_projectGuid, project, targetFrameworkVersion);
+            }
 
-                string? targetFrameworkIdentifier = await configuration.TargetFrameworkIdentifier.GetDisplayValueAsync();
+            static async Task<(string, string)> GetTargetFrameworkAsync(ConfiguredProject project)
+            {
+                var projectProperties = project.Services.ExportProvider.GetExportedValue<ProjectProperties>();
+                var configuration = await projectProperties.GetConfigurationGeneralPropertiesAsync();
 
-                string? targetFrameworkVersion = await configuration.TargetFrameworkVersion.GetDisplayValueAsync();
+                string targetFramework = await configuration.TargetFrameworkIdentifier.GetDisplayValueAsync();
+                string targetFrameworkVersion = await configuration.TargetFrameworkVersion.GetDisplayValueAsync();
 
-                if (!string.Equals(targetFrameworkIdentifier, s_netCoreTargetFrameworkIdentifier, StringComparisons.FrameworkIdentifiers) ||
-                    string.IsNullOrEmpty(targetFrameworkVersion))
-                {
-                    targetFrameworkVersion = string.Empty;
-                }
-                
-                _missingSetupComponentRegistrationService.RegisterPossibleMissingSdkRuntimeVersion(_projectGuid, project, targetFrameworkVersion);
+                return (targetFramework, targetFrameworkVersion);
+            }
+
+            static bool IsNotNetCore(string targetFramework, string targetFrameworkVersion)
+            {
+                return !string.Equals(targetFramework, s_netCoreTargetFrameworkIdentifier, StringComparisons.FrameworkIdentifiers)
+                       || string.IsNullOrEmpty(targetFrameworkVersion);
```

Comment:
Maybe it's just me, but I found the previous code clearer here. What do others think?

Reply:
removed this change